be/src/vec/functions/array/function_array_join.h

```diff
@@ -0,0 +1,261 @@
+                                      : nullptr);
+
+        std::string sep_str = _get_string_from_column(sep_column);
+        std::string null_replace_str = _get_string_from_column(null_replace_column);
+
+        DataTypePtr src_column_type = block.get_by_position(arguments[0]).type;
+        auto nested_type = assert_cast<const DataTypeArray&>(*src_column_type).get_nested_type();
+
+        auto dest_column_ptr = ColumnString::create();
+        DCHECK(dest_column_ptr != nullptr);
+        dest_column_ptr->reserve(input_rows_count);
+
+        auto res_val = _execute_by_type(*src_nested_column, src_offsets, src_null_map, sep_str,
+                                        null_replace_str, nested_type, dest_column_ptr);
+        if (!res_val) {
+            return Status::RuntimeError(
+                    fmt::format("execute failed or unsupported types for function {}({},{},{})",
+                                get_name(), block.get_by_position(arguments[0]).type->get_name(),
+                                block.get_by_position(arguments[1]).type->get_name(),
+                                block.get_by_position(arguments[2]).type->get_name()));
+        }
+
+        block.replace_by_position(result, std::move(dest_column_ptr));
+        return Status::OK();
+    }
+
+private:
+    std::string _get_string_from_column(const ColumnPtr& column_ptr) {
+        if (!column_ptr) {
+            return std::string("");
+        }
+        const ColumnString* column_string_ptr = check_and_get_column<ColumnString>(*column_ptr);
+        StringRef str_ref = column_string_ptr->get_data_at(0);
+        std::string str(str_ref.data, str_ref.size);
+        return str;
+    }
+
+    void _get_result_string(std::vector<std::string>& str_vec, const std::string& sep_str,
+                            std::string& result_str) {
+        for (size_t i = 0; i < str_vec.size(); ++i) {
+            if (i == str_vec.size() - 1) {
+                result_str += str_vec[i];
+            } else {
+                result_str += str_vec[i] + sep_str;
+            }
+        }
+        return;
+    }
+
+    template <typename ColumnType>
+    bool _execute_number(const IColumn& src_column, const ColumnArray::Offsets& src_offsets,
+                         const NullMapType* src_null_map, const std::string& sep_str,
+                         const std::string& null_replace_str, DataTypePtr& nested_type,
+                         ColumnString* dest_column_ptr) {
+        using NestType = typename ColumnType::value_type;
+        bool is_decimal = IsDecimalNumber<NestType>;
+
+        const ColumnType* src_data_concrete = reinterpret_cast<const ColumnType*>(&src_column);
+        if (!src_data_concrete) {
+            return false;
+        }
+
+        ColumnArray::Offset prev_src_offset = 0;
+        for (auto curr_src_offset : src_offsets) {
+            std::string result_str;
+            std::vector<std::string> str_vec;
```

Comment:
append result directly, no need to use str_vec

Reply:
done